### ISSUE TITLE
Restart/Update & Restart Failing (Windows)

### ISSUE THREAD

**Issue by LeeThompson** [nosmokingbandit/Watcher3#183]
If Watcher needs to restart (update or restart command) it fails on Windows, I suspect it's because it's not quoting the file path (which in my case contains spaces).

```
INFO [2017-12-22 14:34:42,583] core.ajax.update_server.453: Update process started.
INFO [2017-12-22 14:34:42,583] CPTaskScheduler.stop.282: Stopping scheduled task Movie Search.
INFO [2017-12-22 14:34:42,583] CPTaskScheduler.stop.282: Stopping scheduled task Update Checker.
INFO [2017-12-22 14:34:42,583] CPTaskScheduler.stop.282: Stopping scheduled task Metadata Update.
INFO [2017-12-22 14:34:42,584] core.version.execute_update.209: Updating from Git.
DEBUG [2017-12-22 14:34:42,584] core.version.fetch.163: Fetching latest Git info.
DEBUG [2017-12-22 14:34:42,584] core.version.runner.95: Executing Git command: ['C:\\Git\\bin\\git.exe', 'fetch']
DEBUG [2017-12-22 14:34:43,942] core.version.pull.173: Pulling latest commit.
DEBUG [2017-12-22 14:34:43,942] core.version.runner.95: Executing Git command: ['C:\\Git\\bin\\git.exe', 'pull']
INFO [2017-12-22 14:34:45,816] core.version.execute_update.225: Update successful.
INFO [2017-12-22 14:34:46,831] cherrypy.error.error.219: [22/Dec/2017:14:34:46] ENGINE Bus STOPPING
INFO [2017-12-22 14:34:46,945] cherrypy.error.error.219: [22/Dec/2017:14:34:46] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('0.0.0.0', 9191)) shut down
INFO [2017-12-22 14:34:46,945] cherrypy.error.error.219: [22/Dec/2017:14:34:46] ENGINE Stopped thread 'Session cleanup'.
INFO [2017-12-22 14:34:46,956] cherrypy.error.error.219: [22/Dec/2017:14:34:46] ENGINE Bus STOPPED
INFO [2017-12-22 14:34:46,957] core.restart.76: Server stopped -- respawning script as: 
 C:\Python3\pythonw.exe C:\Program Files (x86)\Watcher\Watcher3\watcher.py
```

This particular bit here will fail if it's executed exactly as shown:

`C:\Python3\pythonw.exe C:\Program Files (x86)\Watcher\Watcher3\watcher.py`

It likely needs to be:

`C:\Python3\pythonw.exe "C:\Program Files (x86)\Watcher\Watcher3\watcher.py"`

I haven't located the actual bit of code in Watcher that's doing this yet, if I find it, I'll test, and report back with updated code.  (core.restart.76 is not especially helpful since there's no core.py ).

(If folks need a workaround, either moving Watcher to a path without spaces or creating a junction would probably do the trick.)

**UPDATE:**

I moved my Watcher 3 to a new folder structure without spaces in the names and restart now works as expected so I think when Watcher does it's restart it needs to enclose the path in quotes on Windows

**UPDATE:**

Sickrage, Headphones and CouchPotato, which do not have this issue (and also use Python 2.x) use a different call (subprocess.Popen) to restart.   All 3 use similar code:

```
popen_list = []
popen_list = [sys.executable, sickbeard.MY_FULLNAME]
if popen_list and not sickbeard.NO_RESTART:
     popen_list += sickbeard.MY_ARGS
     if '--nolaunch' not in popen_list:
          popen_list += ['--nolaunch']
           subprocess.Popen(popen_list, cwd=os.getcwd())
os._exit(0)
```

```
args = [sys.executable] + [os.path.join(base_path, os.path.basename(__file__))] + sys.argv[1:]
subprocess.Popen(args)
```

```
popen_list = [sys.executable, FULL_PATH]
popen_list += ARGS
subprocess.Popen(popen_list, cwd=os.getcwd())
```

Watcher, of course, is using:

```
python = sys.executable
args = [SCRIPT_PATH] + sys.argv[1:]
os.execl(python, python, *args)
```

**Comment by LeeThompson** [comment]
Update: 

I moved my Watcher 3 to a new folder structure without spaces in the names and restart now works as expected so I think when Watcher does it's restart it needs to enclose the path in quotes on Windows.

**Comment by nosmokingbandit** [comment]
Fixed in 986b52a9da40def45c9fe778d3b3aa6d8fb566dc.
